components/brave_wallet/browser/scrypt_utils.cc

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/scrypt_utils.h"
+
+#include "base/check.h"
+#include "base/containers/span.h"
+#include "base/containers/span_writer.h"
+#include "base/containers/to_vector.h"
+#include "base/numerics/safe_conversions.h"
+#include "brave/vendor/bat-native-tweetnacl/tweetnacl.h"
+
+namespace brave_wallet {
+
+static_assert(kSecretboxNonceSize == crypto_secretbox_NONCEBYTES,
+              "kSecretboxNonceSize must equal crypto_secretbox_NONCEBYTES");
+static_assert(kScryptKeyBytes == crypto_secretbox_KEYBYTES,
+              "kScryptKeyBytes must be equal crypto_secretbox_KEYBYTES");
+
+std::optional<std::vector<uint8_t>> ScryptEncrypt(
+    base::span<const uint8_t> plaintext,
+    base::span<const uint8_t, kScryptKeyBytes> key,
+    base::span<const uint8_t, kSecretboxNonceSize> nonce) {
+  std::vector<uint8_t> padded_plaintext(
+      plaintext.size() + crypto_secretbox_ZEROBYTES, 0);
+  std::vector<uint8_t> ciphertext = padded_plaintext;
```

Comment:
It should be just same size as above
`std::vector<uint8_t> ciphertext(padded_plaintext.size());`